### ISSUE TITLE
Use temporary file name while file download in progress

### ISSUE THREAD

**Issue by andrewwhite72** [aws/aws-cli#701]
I would like to see a feature where a temporary file name is used while aws s3 is downloading a file.
I have a download job that runs many time throughout the day that uses "aws s3 sync" to download files from S3 to my local storage. I then have an ETL job that also runs many times a day that reads the files from the local storage, transforms and loads that data into a data warehouse.
Since the downloaded files appear in a local directory listing before the download is complete there is a chance that the second job will attempt to read a file that has not completely downloaded causing that job to fail.
I would prefer not to have to take steps to prevent these two jobs running concurrently. A simple solution would for aws-cli to provide an option to use a temporary file name for the target while the download is in process (e.g. by prefixing a ".") and then rename the file once the download is complete.

**Comment by kyleknap** [comment]
After switching over the ``s3`` commands to ``s3transfer`` this happens by default for downloads. Closing issue.

**Comment by andrewwhite72** [comment]
So the recommendation is to NOT use the CLI?

**Comment by kyleknap** [comment]
The last comment was not too clear. Let me rephrase:

The CLI now downloads to a temporary file and once the file is completely downloaded, it is then moved to the final location. This is now done by default. So no flag/configuration is required. Therefore closing the issue as it should fulfill the feature request.
